### DISCUSSION TITLE
Any Warranty issues in using proxy mode?

### DISCUSSION THREAD

**Discussion by cliver1956** [johanmeijer/grott#178] [Q&A] [answered]
I am new to the solar world having recently had a system installed consisting of:

    2 Strings of 8 x 410w Hyundai Black Mono Panels
    SPH 6000TL BL-UP 6kW Hybrid Inverter
    GBLI6532 6.5kw Storage Battery
    ShineWiFi-X Data Logger

I also have an IotaWatt power monitor recording power data for my household consumption, PV output and Grid Import and Export power. This data is uploaded to and influxdb running in a docker container on my nas.

The data missing from my influxdb is Battery charge/discharge etc. All stuff that the Shine app shows me (albeit delayed).
I want to find the best way to get the data from my Growatt inverter into either the same database or another on running on my infuxdb instance.

I like the idea of Grott. But, my only reservations are:

    I can't access the nessesary options to use grott in it's preffered proxy mode. It looks like I will get them if I say yes to a disclaimer pop up when I try to access the setup pages. I haven't done so yet because I am uncertain what the warranty implications would be. My installer monitors my "plant" as well. Presumably from the dashboard at server.growatt.com. So I need to make sure they maintain a data feed (in theory they will respond to faults before I know about them ;). Has anyone else had to accept the disclaimer? What are the implications of doing so?
    The other option is to use grott in sniffer mode (packet sniffing at the network level). I understand this doesn't involve any "messing" with the current installation but it is more process intensive and isn't the recommended method). So my concern is if my nas or router will cope with the load efficiently. NAS is an Asustor AS640T, Router Asus RT-AX88U.
Any advice appreciated.

**Comment by johanmeijer** [comment]
I am not sure which disclaimer you means. 

Do mean the growatt pop at the momentyou try to enter the datalogger or inverter settings?

![image](https://user-images.githubusercontent.com/16508298/193650635-25651809-0358-4163-99a5-099eaf6c86e7.png)

That is a default message and not depended on the use of Grott. You will get this message without Grott also.

**Reply by cliver1956** [reply]
Yes, that's the pop up I'm referring to. My question isn't so much about grott but more about my guarantee. In other words, if I accept the disclaimer that pops up and change the server address to my proxy, will it invalidate my warranty?
Especially considering that the installation company also monitors it for support. I'm theory it would be transparent to them as they would get their feed from the growatt server. But, if I had a fault and they discover that I've been "messing" with the setup, would they claim my warranty is void?

**Reply by robupham** [reply]
I’m no lawyer but I think it would be a huge stretch for anyone (installer or Growatt) to claim that changing the IP address that the data logger uses (which itself is an optional component) invalidates the warranty of your inverter.

Only you can decide whether you’re happy to proceed… but personally I had my system installed in mid September, have made the change to use Grott in proxy mode, and everything is fully functional and working fine.

You’re not making material changes to the way the inverter operates (e.g. asking the battery to discharge to an unhealthy level, etc.).

**Reply by cliver1956** [reply]
Thanks Rob, that's how I see it as well. 
I did consider asking my suppliers "permission" but then I thought they might take the easy out and say no anyway. 
I also considered using grott in sniffer mode but I'll be running it in a docker container in my nas so not sure the extra processing load is a good idea.
Think I'll just bite the bullet and go for it :)
Thanks for the confidence boost ;)

**Reply by johanmeijer** [reply]
In addition to @robupham. You can also change IP address during setup of your datalogger (shinewifi / shinelan, shinelink) it is not necessary to do this via the website. 

Grott proxy is really not changing the data. It forward the data 1 to 1 to growatt (in proxy mode) and for forward the responses from growatt to the inverter (unless you have specified blockcmd =True, but that is for your own safety). I do not think that growatt noticed that grott is in between.  But then you have to trust me at my blue eyes (or read the code) ;)

**Reply by cliver1956** [reply]
The problem is the shine WiFi was set up by the installers so I'll have to change it via the web UI.
The blockcmd is certainly an attractive option, or it certainly will be come ww3. Plus, how could I not trust you and your blue eyes? 😉

**Reply by cliver1956** [reply]
Jus a couple of questions before I jump in: Am I reading the wiki right... 
If I only want to send my data to InfluxDB and onwards to the growatt server I don't need MQTT enabled?
If I want to change any perameters post install, such as add blockcmd, I will need to edit my grott.ini file and rerun docker run command?
I am assuming there is no operational user interface once it is up and running?

**Comment by johanmeijer** [comment, marked as answer]
If you only want to use influxdb then MQTT processing can be disabled (nomqtt = True)

If you "externalized" the grott.ini as described in the wikki that is what you can do. If you use the run whit the "g" environmentals you have to "rebuild" the docker run (and remove the container). There is where docker compose comes in to help you (or other docker management tool like portainer or kubernetes). 

There is no user interface.

**Reply by cliver1956** [reply]
Brilliant, many thanks.
